apps/api/src/app/events/usecases/verify-payload/verify-payload.usecase.ts

```diff
@@ -0,0 +1,81 @@
+import { ITemplateVariable } from '@novu/dal';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { VerifyPayloadCommand } from './verify-payload.command';
+
+export class VerifyPayload {
+  execute(command: VerifyPayloadCommand): Record<string, unknown> {
+    const invalidKeys = [];
+    const defaultPayload = {};
+
+    for (const step of command.template.steps) {
+      invalidKeys.push(...this.checkRequired(step.template.variables || [], command.payload));
```

Comment:
This is amazing, but there are a few variables we are passing to the template:
- payload
- subscriber (the current subscriber we are sending it to)
- step (this comes from the digest)

Those variables are injected to the template compilation but are not in the command.payload. So we need to create a combined variables object containing all of the above.

Reply:
The other two variables are currently filled by the app itself and not the user. I would keep the current functionality only to verify the custom event payload and not the fields generated by the app. Wdyt?

Reply:
I would rather mark these variables as system reserved if the user uses them in the template. Because currently the user can just override them and this would case some unexpected issues. Wdyt?

![Screenshot 2022-08-31 230854](https://user-images.githubusercontent.com/13546486/187785242-876aac5d-f44f-4151-accf-0e81d27cd936.png)

Reply:
yes this makes perfect sense. Maybe we can do a less in your face message there, because it's not an error but an info. Maybe doing it with a bright grey color instead of red?

Reply:
Sure, I changed it :)